src/devices/amina.ts

```diff
@@ -0,0 +1,355 @@
+                if ((evStatus & (1 << 3)) !== 0) statusText = 'Charging Paused';
+
+                result.derated = (evStatus & (1 << 15)) !== 0;
+                if (result.derated) statusText += ', Derated';
+
+                result.ev_status = statusText;
+
+                return result;
+            }
+        },
+    } satisfies Fz.Converter,
+
+    alarms: {
+        cluster: 'aminaControlCluster',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result: KeyValue = {};
+
+            if (msg.data.alarms !== undefined) {
+                result.alarms = [];
+                result.alarm_active = false;
+
+                for (let i = 0; i < 16; i++) {
+                    if ((msg.data['alarms'] >> i) & 0x01) {
+                        let alarm = aminaAlarms[i];
+                        if (alarm === undefined) {
+                            alarm = `Unknown Alarm bit #${i}`;
+                        }
+
+                        (result.alarms as string[]).push(alarm);
+                        result.alarm_active = true;
+                    }
+                }
+
+                if (result.alarm_active === false) {
+                    result.alarms = 'No Alarm';
+                }
+
+                return result;
+            }
+        },
+    } satisfies Fz.Converter,
+};
+
+const tzLocal = {
+    charge_limit: {
+        key: ['charge_limit'],
+        convertSet: async (entity, key, value, meta) => {
+            const payload = {level: value, transtime: 0};
+            await entity.command('genLevelCtrl', 'moveToLevel', payload, utils.getOptions(meta.mapped, entity));
+        },
+
+        convertGet: async (entity, key, meta) => {
+            await entity.read('genLevelCtrl', ['currentLevel'], manufacturerOptions);
+        },
+    } satisfies Tz.Converter,
+
+    ev_status: {
+        key: ['ev_status'],
+        convertGet: async (entity, key, meta) => {
+            await entity.read('aminaControlCluster', ['evStatus'], manufacturerOptions);
+        },
+    } satisfies Tz.Converter,
+
+    alarms: {
+        key: ['alarms'],
+        convertGet: async (entity, key, meta) => {
+            await entity.read('aminaControlCluster', ['alarms'], manufacturerOptions);
+        },
+    } satisfies Tz.Converter,
+
+    total_active_power: {
+        key: ['total_active_power'],
+        convertGet: async (entity, key, meta) => {
+            await entity.read('haElectricalMeasurement', ['totalActivePower'], manufacturerOptions);
+        },
+    } satisfies Tz.Converter,
+};
+
+const definitions: DefinitionWithExtend[] = [
+    {
+        zigbeeModel: ['amina S'],
+        model: 'amina S',
+        vendor: 'Amina Distribution AS',
+        description: 'Amina S EV Charger',
+        ota: ota.zigbeeOTA,
+        fromZigbee: [fzLocal.charge_limit, fzLocal.ev_status, fzLocal.alarms, fz.electrical_measurement],
```

Comment:
Could you all move them to extend?

Reply:
Moved most to extend, but not sure how I would do e.text('ev_status' and 'alarms'. Open for suggestions.

Reply:
Looks good!